packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -407,13 +407,44 @@ export class DataStores implements IDisposable {
 		this.contexts.bind(id);
 	}
 
+	/**
+	 * Generate compact internal DataStore ID.
+	 *
+	 * A note about namespace and name collisions:
+	 * This code assumes that that's the only way to generate internal IDs, and that it's Ok for this namespace to overlap with
+	 * user-provided alias names namespace.
+	 * There are two scenarios where it could cause trouble:
+	 * 1) Old files, where (already removed) CreateRoot*DataStore*() API was used, and thus internal name of data store
+	 * was provided by user. Such files may experience name collision with future data stores that receive a name generated
+	 * by this function.
+	 * 2) Much less likely, but if it happen that internal ID (generated by this function) is exactly the same as alias name
+	 * that user might use in the future, them CotnainerRuntime.getAliasedDataStoreEntryPoint() or
+	 * ContainerRuntime.getDataStoreFromRequest() could return a data store with internalID matching user request, even though
+	 * user expected some other data store (that would receive alias later).
+	 * Please note that above mentioned functions have the implementation they have (allowing #2) due to #1.
+	 */
```

Comment:
> Much less likely, but if it happen that internal ID (generated by this function) is exactly the same as alias name that user might use in the future, them CotnainerRuntime.getAliasedDataStoreEntryPoint() or ContainerRuntime.getDataStoreFromRequest() could return a data store with internalID matching user request, even though user expected some other data store (that would receive alias later).

Shouldn't we do something about this? Can't we limit the back-compat code in the existing APIs to only apply to not apply for new containers or something? Like if ID compressor is enabled, don't do back-compat or something?